packages/client/lib/net/peerpool.ts

```diff
@@ -62,11 +63,13 @@ export class PeerPool extends EventEmitter {
     if (this.opened) {
       return false
     }
-    this.config.servers.map((s) => {
-      s.on('connected', (peer: Peer) => {
+    // TODO: Do we still need this map now that we have the centralized event bus?
+    this.config.servers.map(() => {
+      this.config.events.on(Event.PEER_CONNECTED, (peer: any) => {
```

Comment:
do we need this `any` casting on `peer`? I'm seeing it in a lot of places, ideally it would automatically infer the type itself.

If the type is too complicated (looks like it's currently `[Libp2pPeer | RlpxPeer | Devp2pRlpxPeer | MockPeer | Peer]`) could we just simplify it to `Peer` since they all inherit from that?

Reply:
I struggled with this one because it popped up over and over again.  I think I just figured it out after you asked this question but the problem appears to arise when I use that `Devp2pRlpxPeer` type that's borrowed from `devp2p` since I guess that's a different type entirely from `Peer | RlpxPeer | Libp2pPeer` now that I look at it.  It looks like maybe my issue is really where I'm passing that `Devp2pRlpxPeer` with the event (as [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/121a617f94791477f25174c912ba5b066922cc01/packages/client/lib/net/peer/rlpxpeer.ts#L120)).  If I read this right, I think I can safely change `rlpxPeer` to `this` and then I can remove that `Devp2pRlpxPeer` type and that will remove the need for any of this funny typecasting.  Will experiment and let you know.